clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/security/KubernetesNamedAccountCredentials.groovy

```diff
@@ -0,0 +1,206 @@
+import com.netflix.spinnaker.clouddriver.kubernetes.config.LinkedDockerRegistryConfiguration
+import com.netflix.spinnaker.clouddriver.security.AccountCredentials
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsRepository
+import io.fabric8.kubernetes.client.Config
+
+public class KubernetesNamedAccountCredentials implements AccountCredentials<KubernetesCredentials> {
+  static final String CLOUD_PROVIDER = "kubernetes"
+  final String accountName
+  final String environment
+  final String accountType
+  final String context
+  final String cluster
+  final String user
+  final String userAgent
+  final String kubeconfigFile
+  List<String> namespaces
+  KubernetesCredentials credentials
+  final List<String> requiredGroupMembership
+  final List<LinkedDockerRegistryConfiguration> dockerRegistries
+  private final AccountCredentialsRepository accountCredentialsRepository
+
+  public KubernetesNamedAccountCredentials(AccountCredentialsRepository accountCredentialsRepository,
+                                           String userAgent,
+                                           String accountName,
+                                           String environment,
+                                           String accountType,
+                                           String context,
+                                           String cluster,
+                                           String user,
+                                           String kubeconfigFile,
+                                           List<String> namespaces,
+                                           List<LinkedDockerRegistryConfiguration> dockerRegistries,
+                                           List<String> requiredGroupMembership) {
+    if (accountName == null || accountName.isEmpty()) {
+      throw new IllegalArgumentException("Account name for Kubernetes provider missing.")
+    }
+    if (dockerRegistries == null || dockerRegistries.size() == 0) {
+      throw new IllegalArgumentException("Docker registries for Kubernetes account " + accountName + " missing.")
+    }
+    this.accountName = accountName
+    this.environment = environment
+    this.accountType = accountType
+    this.context = context
+    this.cluster = cluster
+    this.user = user
+    this.userAgent = userAgent
+    this.kubeconfigFile = kubeconfigFile != null && kubeconfigFile.length() > 0 ?
+        kubeconfigFile : System.getProperty("user.home") + "/.kube/config"
+    this.namespaces = namespaces
+    this.requiredGroupMembership = requiredGroupMembership == null ? Collections.emptyList() : Collections.unmodifiableList(requiredGroupMembership)
+    this.dockerRegistries = dockerRegistries
+
+    this.accountCredentialsRepository = accountCredentialsRepository
+    this.credentials = buildCredentials()
+  }
+
+  @Override
+  String getName() {
+    return accountName
+  }
+
+  @Override
+  String getCloudProvider() {
```

Comment:
`final String cloudProvider = "kubernetes"` up top would remove the need for this